### PR TITLE
Use package names to import modules and fix lots of typings.

### PR DESCRIPTION
**Pull request by justinfagnani**
This also removes some weird folder structure and the bower_components symlink.

### CODE REVIEW

File: src/lit-element.ts [hunk 1]
```diff
@@ -70,26 +87,26 @@ export function styleString(styleInfo: any) {
...
 
-
-
   /**
    * Returns an
-   * @returns {Node|ShadowRoot} Returns a node into which to render.
-  */
-  protected _createRoot() {
-    return this.attachShadow({mode: 'open'});
+   * @returns {Element|DocumentFragment} Returns a node into which to render.
```
Comment (rictic) [inline src/lit-element.ts]: Types can (and IMO should) be left out of jsdoc in ts files

File: src/lit-element.ts [hunk 2]
```diff
@@ -112,13 +129,13 @@ export class LitElement extends PropertiesMixin(HTMLElement) {
    * @param {*} changedProps Changing element properties
    * @param {*} prevProps Previous element properties
    */
-  protected _propertiesChanged(props: any, changedProps: any, prevProps: any) {
+  _propertiesChanged(props: any, changedProps: any, prevProps: any) {
     this.__isChanging = true;
     this.__isInvalid = false;
     super._propertiesChanged(props, changedProps, prevProps);
     const result = this.render(props);
```
Comment (rictic) [inline src/lit-element.ts]: Where is render defined? I don't see it anywhere here.

File: src/lit-element.ts [hunk 3]
```diff
@@ -101,7 +118,7 @@ export class LitElement extends PropertiesMixin(HTMLElement) {
    * @param {*} _prevProps Previous element properties
    * @returns {boolean} Default implementation always returns true.
    */
-  protected _shouldPropertiesChange(_props: any, _changedProps: any, _prevProps: any) {
+  _shouldPropertiesChange(_props: any, _changedProps: any, _prevProps: any) {
```
Comment (rictic) [inline src/lit-element.ts]: Prefer `object` to `any`, here and on `_propertiesChanged`

File: src/lit-element.ts (hunk 1 above)
Comment (justinfagnani) [reply src/lit-element.ts]: done

File: src/lit-element.ts (hunk 2 above)
Comment (justinfagnani) [reply src/lit-element.ts]: it's in here... it's abstract-ish

File: src/lit-element.ts (hunk 3 above)
Comment (justinfagnani) [reply src/lit-element.ts]: done here and elsewhere

**Comment by kevinpschaaf** [comment]
@justinfagnani @sorvell Sorry after the fact, but would it make sense to add a note to the readme that points out previous versions of lit-html used path-based references to dependencies, current version uses names, and thus name-to-path conversion is necessary before loading in the browser (with pointers to polymer-cli and/or other tools)?  Seems worth pointing out the change maybe?

**Comment by samuelli** [comment]
The readme is also incorrect now, right?

**Pull request merged**